src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Validators/TimePeriodStringValidators.cs

```diff
@@ -14,8 +14,13 @@ public static partial class TimePeriodStringValidators
 
     public static IRuleBuilderOptionsConditions<T, string> TimePeriodString<T>(this IRuleBuilder<T, string> rule)
     {
-        return rule.Custom((value, context) =>
+        return rule.NotEmpty().Custom((value, context) =>
         {
+            if (value.IsNullOrWhitespace())
+            {
+                return;
+            }
```

Comment:
This is to prevent FluentValidation's somewhat annoying default cascading behaviour. Without this, if the user provides an empty string in the query parameter, it gets mapped to a null in the request model.

The null will fail the `NotEmpty` validation, but cascading behaviour will mean that the null still comes through to our custom validator. When the null hits the format validation, it then throws an exception, resulting in a 500 response.

It's possible to change FluentValidation's cascading behaviour, but this isn't very consistent when combined with `RuleForEach`. Unfortunately, this stops validating on the first item that has an error in the list, rather than report errors for every item in the list.

It's also annoying to have to remember to set the cascade behaviour manually. Consequently, I've opted for this approach as it avoids all the above issues.

Reply:
I see. That makes sense